bootstrap_cfn/config.py

```diff
@@ -525,6 +526,8 @@ def rds(self, template):
             ],
             VpcId=Ref("VPC"),
             GroupDescription="SG for EC2 Access to RDS",
+            # translates to DependsOn="AttachGateway",
+            DependsOn=[r.title for r in self._find_resources(template, "AWS::EC2::VPCGatewayAttachment")],
```

Comment:
I 've made it like this in another commit 
```python
            DependsOn=[
                r.title
                for r in self._find_resources(
                    template, "AWS::EC2::VPCGatewayAttachment")],
```
Alternatively we can do 
```
    @classmethod
    def _filter_list_by_attr(cls, input_list, attr_name, attr_value):
        filtered = (r for r in input_list
                if hasattr(r, attr_name) and
                getattr(r, attr_name) == attr_value)
        return filtered
```
though we would only benefit from that for the ScalingGroup ... 

Another approach would be to do this via a different function, in the same fashion we use `_attach_elbs`, that way we can find the resources near the end and update the respective DependsOn attributes. This third method is probably the cleanest way but I 'd rather we get this in and refactor later.